docs/source/integrations/iceberg_versioning.md

```diff
@@ -0,0 +1,228 @@
+
+```bash
+pyiceberg[pyarrow]
+```
+Depending on your choice of storage, you may also need to install the optional dependencies. Consult [the installation guide for `PyIceberg`](https://py.iceberg.apache.org/#installation) to update the line above with the necessary optional dependencies.
+
+Now you can install the project requirements with the following command:
+```
+pip install -r requirements.txt
+```
+### Set up the Iceberg catalog
+
+Iceberg tables are managed by a catalog which is responsible for managing the metadata of the tables. In production, this could be a Hive, Glue, or [other catalog supported by Apache Iceberg](https://py.iceberg.apache.org/configuration/#catalogs). Iceberg also supports various storage options such as S3, HDFS, and more. There are multiple ways you can configure the catalog, credentials, and object storage to suit your needs by refering to the [configuration guide](https://py.iceberg.apache.org/configuration/). For this tutorial, we will use the `SQLCatalog` which stores the metadata in a local `sqlite` database and uses the local filesystem for storage.
+
+Create a temporary location for Iceberg tables by running the following command:
+
+```bash
+mkdir -p /tmp/warehouse
+```
+Then create or update a file called `.pyiceberg.yaml` in the root directory of the system with the following content:
+
+```yaml
+catalog:
+  default:
+    type: sql
+    uri: sqlite:////tmp/warehouse/pyiceberg_catalog.db
+    warehouse: file:///tmp/warehouse/warehouse
+```
+You can check if the configuration is by opening a Python shell with `ipython` command and running the following code:
+
+```python
+from pyiceberg.catalog import load_catalog
+catalog = load_catalog(name="default")
+```
+## Define a custom dataset to use Iceberg tables
+
+To use the Iceberg tables with Kedro, you will need to define a [custom dataset that uses the `pyiceberg` library](../data/how_to_create_a_custom_dataset.md). Create a new file  called `pyiceberg_dataset.py` in the `src/<package_name>` directory of your project and copy the following code:
+
+```python
+import pyarrow as pa
+from kedro.io.core import AbstractDataset, DatasetError
+from pyiceberg.catalog import load_catalog
+from pyiceberg.exceptions import NoSuchTableError
+
+DEFAULT_LOAD_ARGS = {"load_version": None}
+DEFAULT_SAVE_ARGS = {"mode": "overwrite"}
+
+class PyIcebergDataset(AbstractDataset):
+    def __init__(
+            self,
+            catalog,
+            namespace,
+            table_name,
+            table_type="pandas",
+            load_args=DEFAULT_LOAD_ARGS,
+            scan_args=None,
+            save_args=DEFAULT_SAVE_ARGS,
+    ):
+        self.table_name = table_name
+        self.namespace = namespace
+        self.catalog = load_catalog(catalog)
+        self.table_type = table_type
+        self.load_args = load_args
+        self.table = self._load_table(namespace, table_name)
```

Comment:
Feels weird that this lookup is happening in `__init__()`; have you tested this for a table that will be created during the run?

Reply:
Yeah, when saving, it checks if the table exists, and if it doesn't, it creates one. This load would help to access the metadata